### PR TITLE
Add cloud model cache doc

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

* Add cloud model cache doc.
* Fix typo.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: docs/source/features/cloud_model_cache.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Cloud Model Cache
+
+## What is Cloud Model Cache
+
+The Cloud Model Cache is a system where Olive stores output models in Azure Blob Storage. Whenever a Pass is executed, Olive checks if the same input model has been processed before and if the corresponding output model is available in Azure Blob Storage. If the output model is found, Olive directly uses it for the next step. With this feature, you can easily share output models with others.
```
Comment (devang-ml) [inline docs/source/features/cloud_model_cache.md:5]: stores "intermediate" models

File: docs/source/features/cloud_model_cache.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+This feature only supports native Huggingface models as input.
```
Comment (devang-ml) [inline docs/source/features/cloud_model_cache.md:7]: I'd add "currently"

File: docs/source/features/cloud_model_cache.md (hunk 1 above)
Comment (xiaoyu-work) [reply docs/source/features/cloud_model_cache.md]: updated.

File: docs/source/features/cloud_model_cache.md (hunk 2 above)
Comment (xiaoyu-work) [reply docs/source/features/cloud_model_cache.md]: updated

File: docs/source/features/cloud_model_cache.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  * `account_url [str]`: The account url from the Azure Storage account.
+  * `container_name [str]`: The container name to store model cache.
+  * `upload_to_cloud [bool]`: Whether to upload output model to the cloud cache for this workflow run. Default by `True`.
+
+Here is an example configuration:
+
+```json
+{
+    "engine: {
+        ...
```
Comment (jambayk) [inline docs/source/features/cloud_model_cache.md:46]: You can use `//...` instead to pass the doc build

Comment (jambayk) [reply docs/source/features/cloud_model_cache.md:46]: Also, `"engine"` is no longer required. The engine config params can be passed directly to the top level dict.

**Review by jambayk — APPROVED** [review]

**Pull request merged**